src/manifold/src/csg_tree.cpp

```diff
@@ -210,26 +210,14 @@ Manifold::Impl CsgLeafNode::Compose(
               combined.halfedge_.begin() + nextEdge,
               UpdateHalfedge({nextVert, nextEdge, nextTri}));
 
-    // Assign new IDs to triangles added in this iteration, to differentiate
```

Comment:
The `compose` function is applying transformations lazily, so different meshes may have the same meshID. Consider composing `A` and `A.Translate(...)`, we will add vertices in `A` two times, but in the second time we apply a transformation while copying the vertices. Not sure if it is sound to just perform one meshID update at the end.

Reply:
This is an excellent point, I clearly didn't think this through. Will fix.

Reply:
Fixed.